### ISSUE TITLE
VBench evaluation results significantly deviate from paper-reported scores (with modified code for compatibility)

### ISSUE THREAD

**Issue by 740316Lyx** [DoHunLee1/VideoGuide#4]
I have been trying to reproduce the results reported in the paper using the provided codebase. After setting up the conda environment and downloading the required models, I found that the original code failed to run due to some environment/version mismatches. I made minor modifications to the code that do not affect the computational logic or output quality (e.g., adjusting import paths, fixing deprecated function calls) to make it executable. However, when I evaluate the generated MP4 videos using VBench, the obtained scores are substantially different from those claimed in the paper.

Environment
GPU: NVIDIA RTX A6000 
others are you provided in github

Run the inference command:
export CUDA_LAUNCH_BLOCKING=1
name="videoguide"

animatediff_ckpt='/mnt/nfs0/linshibo/model/stable-diffusion-v1-5'
ckpt='/mnt/nfs0/linshibo/model/model.ckpt'
config='./animatediff_configs/prompts/v2/v2-1-w-dreambooth.yaml'
vc_config='./vc_configs/inference_t2v_512_v2.0.yaml'
prompts="./prompts/prompt.txt"

python t2v_vc_guide.py \
    --seed 42 \
    --video_length 8 \
    --fps 8 \
    --cfg_scale 0.8 \
    --animatediff_model_path $animatediff_ckpt \
    --vc_model_path $ckpt \
    --config $config \
    --vc_config $vc_config \
    --num_step 50 \
    --savedir "./result" \
    --precision 'float32' \
    --prompt "$prompts" \
    --mode 0 \
    --cfg_plus

<img width="1907" height="721" alt="Image" src="https://github.com/user-attachments/assets/e864e99e-09ed-43aa-88bb-d2ead56d631f" />

vbench_test.py:#!/usr/bin/env python3
from __future__ import annotations

import argparse
import csv
import json
import os
import shutil
import subprocess
import sys
from pathlib import Path

DEFAULT_DIMENSIONS = [
    "subject_consistency",
    "background_consistency",
    "motion_smoothness",
    "dynamic_degree",
    "aesthetic_quality",
    "imaging_quality",
]

def parse_args() -> argparse.Namespace:
    parser = argparse.ArgumentParser(description="Score GIF outputs with local VBench.")
    parser.add_argument(
        "--input_root",
        type=Path,
        default=Path("/home/linshibo/fuxian/VideoGuide/result"),
        help="Root directory that contains generated GIFs.",
    )
    parser.add_argument(
        "--output_root",
        type=Path,
        default=Path("/home/linshibo/fuxian/VideoGuide/result/vbench_gif_scores"),
        help="Directory to store VBench outputs and per-GIF score files.",
    )
    parser.add_argument(
        "--vbench_root",
        type=Path,
        default=Path("/home/linshibo/STG-DiT/VBench"),
        help="Local VBench repository path.",
    )
    parser.add_argument(
        "--cache_dir",
        type=Path,
        default=Path("/home/linshibo/STG-DiT/vbench_cache"),
        help="Local VBench cache directory.",
    )
    parser.add_argument(
        "--dimensions",
        nargs="+",
        default=DEFAULT_DIMENSIONS,
        choices=DEFAULT_DIMENSIONS,
        help="VBench dimensions to evaluate.",
    )
    parser.add_argument(
        "--keep_staging",
        action="store_true",
        help="Keep the flattened temporary GIF directory.",
    )
    parser.add_argument(
        "--allow_download",
        action="store_true",
        help="Allow VBench to download missing checkpoints. Disabled by default.",
    )
    return parser.parse_args()

def safe_name(text: str) -> str:
    return text.replace(os.sep, "__").replace("/", "__")

def collect_gifs(root: Path) -> list[Path]:
    exts = { ".mp4"}
    # 排除这些子目录（可根据需要添加）
    exclude = {"_staging", "vbench_scores", "vbench_gif_scores"}
    videos = []
    for path in root.rglob("*"):
        if path.is_file() and path.suffix.lower() in exts:
            # 如果路径的任意部分在排除列表中，则跳过
            if any(part in exclude for part in path.parts):
                continue
            videos.append(path)
    return videos

def prepare_staging_dir(output_root: Path) -> Path:
    staging = output_root / "_staging"
    if staging.exists():
        shutil.rmtree(staging)
    staging.mkdir(parents=True, exist_ok=True)
    return staging

def flatten_gifs(input_root: Path, staging_dir: Path, videos: list[Path]):
    prompt_map: dict[str, str] = {}
    meta: list[dict[str, str]] = []

    for idx, src in enumerate(videos):
        rel = src.relative_to(input_root)
        rel_stem = rel.with_suffix("")
        staged_name = f"{idx:05d}__{safe_name(str(rel_stem))}{src.suffix.lower()}"
        staged_path = staging_dir / staged_name

        try:
            os.symlink(src, staged_path)
        except OSError:
            shutil.copy2(src, staged_path)

        prompt = rel.parts[0] if len(rel.parts) > 1 else rel_stem.name
        model = rel.stem
        prompt_map[staged_name] = prompt
        meta.append(
            {
                "source_path": str(src),
                "relative_path": str(rel),
                "staged_path": str(staged_path),
                "staged_name": staged_name,
                "prompt": prompt,
                "model": model,
            }
        )

    return prompt_map, meta

def required_cache_paths(cache_dir: Path, dimensions: list[str]) -> list[Path]:
    req = {
        "subject_consistency": [
            cache_dir / "dino_model/facebookresearch_dino_main",
            cache_dir / "dino_model/dino_vitbase16_pretrain.pth",
        ],
        "background_consistency": [cache_dir / "clip_model/ViT-B-32.pt"],
        "motion_smoothness": [cache_dir / "amt_model/amt-s.pth"],
        "dynamic_degree": [cache_dir / "raft_model/models/raft-things.pth"],
        "aesthetic_quality": [
            cache_dir / "clip_model/ViT-L-14.pt",
            cache_dir / "aesthetic_model/emb_reader/sa_0_4_vit_l_14_linear.pth",
        ],
        "imaging_quality": [cache_dir / "pyiqa_model/musiq_spaq_ckpt-358bb6af.pth"],
    }
    paths: list[Path] = []
    for dim in dimensions:
        paths.extend(req.get(dim, []))
    return paths

def fail_if_cache_missing(cache_dir: Path, dimensions: list[str]) -> None:
    missing = [path for path in required_cache_paths(cache_dir, dimensions) if not path.exists()]
    if missing:
        lines = "\n".join(f"  - {path}" for path in missing)
        raise FileNotFoundError(
            "Strict offline mode is enabled and these VBench cache items are missing:\n"
            f"{lines}"
        )

def forbid_download_commands() -> None:
    original_run = subprocess.run

    def guarded_run(cmd, *args, **kwargs):
        argv = cmd if isinstance(cmd, (list, tuple)) else [str(cmd)]
        head = Path(str(argv[0])).name if argv else ""
        if head in {"wget", "curl", "git"}:
            raise RuntimeError(
                "Blocked a download command in strict offline mode: "
                + " ".join(str(part) for part in argv)
            )
        return original_run(cmd, *args, **kwargs)

    subprocess.run = guarded_run

def build_score_index(raw_results: dict) -> dict[str, dict[str, float]]:
    per_video: dict[str, dict[str, float]] = {}
    for dimension, value in raw_results.items():
        if not isinstance(value, list) or len(value) != 2:
            continue
        _, details = value
        for item in details:
            path = item["video_path"]
            per_video.setdefault(path, {})[dimension] = float(item["video_results"])
    return per_video

def write_outputs(raw_results: dict, meta: list[dict[str, str]], output_root: Path):
    output_root.mkdir(parents=True, exist_ok=True)
    per_gif_dir = output_root / "per_gif"
    per_gif_dir.mkdir(parents=True, exist_ok=True)

    score_index = build_score_index(raw_results)
    rows = []

    for item in meta:
        scores = score_index.get(item["staged_path"], {})
        record = {**item, "scores": scores}
        rows.append(record)

        out_name = safe_name(item["relative_path"]) + ".json"
        (per_gif_dir / out_name).write_text(
            json.dumps(record, ensure_ascii=False, indent=2) + "\n",
            encoding="utf-8",
        )

    (output_root / "per_gif_scores.json").write_text(
        json.dumps(rows, ensure_ascii=False, indent=2) + "\n",
        encoding="utf-8",
    )

    dimensions = sorted({dim for row in rows for dim in row["scores"].keys()})
    with (output_root / "per_gif_scores.csv").open("w", encoding="utf-8", newline="") as f:
        writer = csv.writer(f)
        writer.writerow(["prompt", "model", "relative_path", "source_path", *dimensions])
        for row in rows:
            writer.writerow(
                [
                    row["prompt"],
                    row["model"],
                    row["relative_path"],
                    row["source_path"],
                    *[row["scores"].get(dim, "") for dim in dimensions],
                ]
            )

    dimension_summary = {
        dim: raw_results[dim][0] if isinstance(raw_results.get(dim), list) and raw_results[dim] else None
        for dim in raw_results
    }
    (output_root / "dimension_summary.json").write_text(
        json.dumps(dimension_summary, ensure_ascii=False, indent=2) + "\n",
        encoding="utf-8",
    )

def main() -> None:
    args = parse_args()
    input_root = args.input_root.expanduser().resolve()
    output_root = args.output_root.expanduser().resolve()
    vbench_root = args.vbench_root.expanduser().resolve()
    cache_dir = args.cache_dir.expanduser().resolve()

    os.environ["VBENCH_CACHE_DIR"] = str(cache_dir)
    if not args.allow_download:
        os.environ["HF_HUB_OFFLINE"] = "1"
        os.environ["TRANSFORMERS_OFFLINE"] = "1"
        os.environ["HF_DATASETS_OFFLINE"] = "1"
        fail_if_cache_missing(cache_dir, args.dimensions)
        forbid_download_commands()

    sys.path.insert(0, str(vbench_root))

    import torch
    from vbench import VBench
    from vbench.distributed import print0

    if not torch.cuda.is_available():
        raise RuntimeError("这个脚本默认按 GPU 跑 VBench；当前环境没有可用 CUDA。")

    videos = collect_gifs(input_root)
    if not videos:
        raise FileNotFoundError(f"没有在 {input_root} 下找到 GIF/MP4 文件")

    staging_dir = prepare_staging_dir(output_root)
    prompt_map, meta = flatten_gifs(input_root, staging_dir, videos)

    evaluator = VBench(torch.device("cuda"), str(vbench_root / "vbench" / "VBench_full_info.json"), str(output_root))
    raw_name = "vbench_gif_scores"
    raw_path = output_root / f"{raw_name}_eval_results.json"

    evaluator.evaluate(
        videos_path=str(staging_dir),
        name=raw_name,
        prompt_list=prompt_map,
        dimension_list=list(args.dimensions),
        local=True,
        read_frame=False,
        mode="custom_input",
    )

    if not raw_path.is_file():
        raise FileNotFoundError(f"没有找到 VBench 输出文件：{raw_path}")

    raw_results = json.loads(raw_path.read_text(encoding="utf-8"))
    write_outputs(raw_results, meta, output_root)
    print0(f"raw results: {raw_path}")
    print0(f"summary dir: {output_root}")

    if not args.keep_staging:
        shutil.rmtree(staging_dir, ignore_errors=True)

if __name__ == "__main__":
    main()
